### PR TITLE
Added TurnOnIdle for Tiberian Sun infantry

### LINKED ISSUE #14808: Turn when idle trait for infantry
Think nobody remembered this, but in RA2 and TS infantry turned when they were idle (changed their current facing), idk if they did that back in td / ra, but I am certain tanks in d2k also aimed their turrets randomly when idle.

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/OpenRA/OpenRA/issues/14808.

### CODE REVIEW

**Review by abcdefg30 — COMMENTED** [review]
What is the idea behind `turnTicks`? Couldn't we just check if we reached `targetFacing` directly instead?

**Comment by Mailaender** [comment]
I assume it smoothes out the turn.

**Comment by Dzierzan** [comment]
This trait would be helpful in my mod too as units should turn when idle.

**Comment by Mailaender** [comment]
You already have https://github.com/Dzierzan/SwarmAssault/blob/master/OpenRA.Mods.Swarm_Assault/Traits/Behaviour/Living.cs

**Comment by Dzierzan** [comment]
Yeah, but this way the mod would have been less based on custom traits.

File: OpenRA.Mods.Common/Traits/TurnOnIdle.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+				return;
+			}
+
+			if (turnTicks == -1)
+			{
+				targetFacing = self.World.SharedRandom.Next(256);
+				turnTicks = 1 + System.Math.Abs(mobile.Facing - targetFacing) / mobile.TurnSpeed;
+			}
+
+			currentDelay = self.World.SharedRandom.Next(Info.MinDelay, Info.MinDelay);
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/TurnOnIdle.cs]: Should probably be

```suggestion
			currentDelay = self.World.SharedRandom.Next(Info.MinDelay, Info.MaxDelay);
```

**Comment by Mailaender** [comment]
Rebased, removed `turnTicks` and added checks if `mobile` is enabled.

**Comment by Smittytron** [comment]
This works but the infantry look awful jittery turning every 40-80 ticks.  I'd try 5, maybe even 10x longer.

**Comment by Mailaender** [comment]
Increased the delay.

**Review by Smittytron — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/3eb5b0d7d12ec571d81c0aaf2148169571210420...5c7a014631677ed65a242ab9ce829d73a689a098)

**Pull request merged**